include/ginkgo/core/distributed/matrix.hpp

```diff
@@ -123,7 +141,7 @@ template <template <typename, typename> class MatrixType, typename... Args>
 auto with_matrix_type(Args&&... create_args)
 {
     return detail::MatrixTypeBuilderFromValueAndIndex<MatrixType, Args...>{
-        std::make_tuple(create_args...)};
+        std::tuple<Args...>{std::forward<Args>(create_args)...}};
```

Comment:
Args pack members are most likely a reference type here - is that intended?

Reply:
The result of `make_tuple` won't be `tuple<Args...>` but something else. I'm now changing it a bit to be able to use make_tuple. That would be a bit nicer in c++17...

Reply:
are you looking for `https://en.cppreference.com/w/cpp/utility/tuple/forward_as_tuple`?

Reply:
What I mean here: The reference collapsing rules lead to Args binding to the most suitable reference types, so you are passing reference types to the arguments to std::tuple, is that what you intended? Do you want to pass the parameters by lvalue reference (std::tie), by value (std::make_tuple) or by the most suitable kind of reference (std::forward_as_tuple)?

Reply:
I think passing references here is fine. This is only called in the Matrix constructor and `create` is called immediately on the result, so the references will stay valid. Then `std::forward_as_tuple` seems to be the best choice.